### PR TITLE
FileDebugMode

### PR DESCRIPTION
**Pull request by vody105**

### CODE REVIEW

File: src/ExtraConfigurator.php [hunk 1]
```diff
@@ -15,9 +14,6 @@ class ExtraConfigurator extends Configurator
 	public function __construct()
```
Comment (f3l1x) [inline src/ExtraConfigurator.php]: Empty constructor?

File: .docs/README.md [hunk 2]
```diff
@@ -7,94 +7,71 @@
...
-Environment variable must follow this pattern: `NETTE__{NAME}`, `NETTE__{NAME}__{NAME2}`.
+We added two methods to help you detect debug mode. You can either manage debug mode via `NETTE_DEBUG` environmental variable and detect it this way:
 
-```bash
-export NETTE__DATABASE__USER=test
-export NETTE__DATABASE__HOST=localhost
+```php
+$configurator = new ExtraConfigurator();
+$configurator->setEnvDebugMode();
 ```
```
Comment (f3l1x) [inline .docs/README.md:24]: TRUE; -> TRUE.

File: src/ExtraConfigurator.php [hunk 3]
```diff
@@ -69,37 +65,35 @@ public function setEnvDebugMode()
...
-	 * Setup debug mode and add parameters immediately
-	 *
-	 * @param Configurator $configurator
-	 * @return Configurator
+	 * @param string $fileName
+	 * @return void
 	 */
-	public static function setup(Configurator $configurator)
+	public function setFileDebugMode($fileName = NULL)
 	{
```
Comment (f3l1x) [inline src/ExtraConfigurator.php:64]: It's nice. What about little tunning?

```
function setFileDebugMode($fileName = null)
{
	$appDir = $this->parameters['appDir'] ? $this->parameters['appDir'] : null;

	// No filename and no application root dir
	if (!$fileName && !$appDir) return;

	$content = @file_get_contents($fileName ?: $appDir . '/../.debug');

        // Not exists or not readable
        if ($content === false) return;

	if (empty($content)) {
		$this->setDebugMode(true);
	} else {
		$debug = self::parseDebugValue(trim($content));
		$this->setDebugMode($debug);
	}

}
```

Comment (vody105) [reply src/ExtraConfigurator.php:64]: @f3l1x seems fine besides `empty($content)` cant be used cause its invalid for "0" etc

Comment (f3l1x) [reply src/ExtraConfigurator.php:64]: I see, `$content === ''` would be fine.

**Comment by f3l1x** [comment]
Najs.

**Pull request merged**